lib/mutant/isolation.rb

```diff
@@ -47,11 +47,18 @@ def self.call(&block)
             end
 
             writer.close
-            Marshal.load(reader.read)
+            result = Marshal.load(reader.read)
+            if result.is_a?(Mutant::Result::Test)
```

Comment:
I think that this is a very weak test. Better have a dedicated "subprocess result object" that wraps A) a valid test result, B) an exception (with backtrace!)

See Haskells `Either` type or sum types in general.